DOCUMENTRETRIEVAL/documentRetrieval/library/postgresql.py

```diff
@@ -15,21 +15,18 @@ def __init__(self, host="127.0.0.1", port="5432"):
         self.port = port
 
 
-    ## THIS IS NOT SAFE FROM SQL INJECTION ATTACKS. WE CANNOT DINAMICALLY CREATE SQL STATEMENTS AND 
-    ## EXECUTE THEM iN THIS WAY. WE HAVE TO DO THIS THE FOLOWING WAY:
-    ## cursor.execute('SELECT * FROM documents WHERE document_id=%s', (doc_id,))
-
-    def db_query(self,query_words):
+    def db_query(self,query_words): 
         """ From database returns list of dictionaries containing document IDs and text. Documents contain at least one query word.
         Args:
             query_words(list): List of query words
         Returns: 
             documents(list): list of dictionaries containing document IDs and text"""
         output = '|'.join(query_words)
-        SQL = """
-                SELECT document_id, fulltext_cleaned FROM documents
-                WHERE to_tsvector('english', fulltext_cleaned) @@ to_tsquery(""" + '\''+ output + '\');'
-        documents = self.execute(SQL)
+       
+        documents=  self.execute("""
```

Comment:
Move the statement outside of the function. Use multiline string formatting tricks.
Multiline strings: https://www.techbeamers.com/python-multiline-string/
String formatting: https://pyformat.info/

```python
statement = (
    "SELECT document_id, fulltext_cleaned FROM documents "
    "WHERE to_tsvector('english', fulltext_cleaned) @@ to_tsquery({output});"
).format(output=output)

# execute the statement
documents=  self.execute(statement)
```

Reply:
Still you should do this the following way: 
```
statement = (
    "SELECT document_id, fulltext_cleaned FROM documents "
    "WHERE to_tsvector('english', fulltext_cleaned) @@ to_tsquery(%s);"
)
# execute the statement
documents=  self.execute(statement, (output,))
```
Using execute in this way, it automatically checks that "output" does not contaion malicious SQL code.